daemons/ipa-slapi-plugins/ipa-otp-lasttoken/ipa_otp_lasttoken.c

```diff
@@ -105,6 +105,41 @@ static bool sdn_is_only_enabled_token(Slapi_DN *target_sdn, const char *user_dn)
     return result;
 }
 
+static bool is_otp_enabled(const char *user_dn)
+{
+    char *attrs[] = { "ipaUserAuthType", NULL };
+    Slapi_Entry *entry = NULL;
+    Slapi_DN *sdn;
+    const Slapi_DN *base;
+    uint32_t authtypes;
+    int search_result = 0;
+    char *authConfigDN;
+
+    sdn = slapi_sdn_new_dn_byval(user_dn);
```

Comment:
You have to check the result of ``slapi_sdn_new_dn_byval`` for NULL. The value may be NULL in case DS is running out of memory or something else goes wrong

```
sdn = slapi_sdn_new_dn_byval(user_dn);
if (sdn == NULL) {
    LOG_TRACE("File '%s' line %d: Unable to convert '%s' to Slapi_DN. ", __FILE__, __LINE__, user_dn);
    return false;
}
```